assets/check

```diff
@@ -5,25 +5,31 @@
 set -e
 
 # parse incoming config data
-payload=`cat`
-bucket=$(echo "$payload" | jq -r '.source.bucket')
-prefix="$(echo "$payload" | jq -r '.source.path // ""')"
+eval "$(jq -r '.source | [
```

Comment:
The very short form of the argument is that jq's `@sh` is trustworthy because it doesn't try to be clever. :)

I've reported more than my share of shell injection bugs in code (mostly in Java) that tried to do eval-safe escaping and got it wrong, but in all those cases the tools in question were trying to do the minimum quoting necessary, leaving trivial content unquoted or using double quotes instead of single quotes if there were literal single quotes in the string to be escaped, or otherwise attempting to be clever.

The POSIX sh rules for single-quoted strings, though, are dirt simple: _Absolutely everything_ is literal, including backslashes, until the next single quote; no mechanism to escape them is allowed. To insert a literal single quote character, jq exits the single-quoted string, adds a backslash-escaped single quote, and then reenters the single-quoted context.

Dirt simple, next to impossible to get wrong.

Reply:
That said -- if y'all are willing to give up `sh` compatibility to switch to `bash`, there are options that become available to drop `eval`, avoid shell injection, _and_ still run jq only once; something like:

```
declare -A source_vars=( )
while IFS='' read -r -d '' element; do
  [[ $element = *=* ]] || continue
  source_vars[${element%%=*}]=${element#*=}
done < <(jq -j '
  def clean_nuls: . | sub("\u0000"; "<NUL>");
  .source | to_entries[] | "\(.key|clean_nuls)=\(.value|clean_nuls)\u0000"
')
```

after which we work with `${source_vars[bucket]}` and the like. That way we import every defined variable, but unexpected values like `PATH`, `LD_PRELOAD`, &c can't be used to attack the system -- one needs to explicitly set `AWS_ACCESS_KEY_ID=${source_vars[access_key_id]}` to get them out of the array.

Reply:
Thanks for the explanation @charles-dyfis-net ! Ultimately, given that the only benefit of these changes is to eliminate duplicate calls to `jq`, I'm inclined not to introduce the use of `eval`. While the changes may be safe in their current form as you say, if `jq` changes it `@sh` behavior, that could change. Or, perhaps more likely, other maintainers of this code (myself included) who don't understand the risks of `eval` as well could introduce a shell-injection risk. 

So I'd like to remove the changes to use `eval`

Reply:
Updated per request. Let me know if you want the change and its revert squashed together.

I do plan on a separate PR to use xargs to remove the eval for options -- might come back with a concrete eval-free single-jq-invocation config-parsing implementation then.